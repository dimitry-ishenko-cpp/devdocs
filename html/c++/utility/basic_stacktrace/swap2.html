    <h1 id="firstHeading" class="firstHeading">std::swap<small>(std::basic_stacktrace)</small>
</h1>            <table class="t-dcl-begin"> <tr class="t-dsc-header"> <th> Defined in header <code><a href="../../header/stacktrace.html" title="cpp/header/stacktrace">&lt;stacktrace&gt;</a></code> </th> <th> </th> <th> </th> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class Allocator &gt;
void swap( std::basic_stacktrace&lt;Allocator&gt;&amp; lhs, std::basic_stacktrace&lt;Allocator&gt;&amp; rhs )
    noexcept(noexcept(lhs.swap(rhs)));</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>Specializes the <code><a href="../../algorithm/swap.html" title="cpp/algorithm/swap">std::swap</a></code> algorithm for <code>std::basic_stacktrace</code>. Swaps the contents of <code>lhs</code> and <code>rhs</code>. Equivalent to <code>lhs.swap(rhs);</code>.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> lhs, rhs </td> <td> - </td> <td> stacktraces whose contents to swap </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>(none)</p>
<h3 id="Complexity"> Complexity</h3> <p>Constant.</p>
<h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap.html" title="cpp/utility/basic stacktrace/swap"> <span class="t-lines"><span>swap</span></span></a></div> </td> <td> swaps the contents <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/basic_stacktrace/swap2" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/basic_stacktrace/swap2</a>
  </p>
</div>
