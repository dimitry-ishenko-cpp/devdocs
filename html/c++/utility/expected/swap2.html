    <h1 id="firstHeading" class="firstHeading">swap<small>(std::expected)</small>
</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">friend constexpr void swap( expected&amp; lhs, expected&amp; rhs ) noexcept(/*see below*/);</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>Overloads the <code><a href="../../algorithm/swap.html" title="cpp/algorithm/swap">std::swap</a></code> algorithm for <code>std::expected</code>. Exchanges the state of <code>lhs</code> with that of <code>rhs</code>. Effectively calls <code>lhs.swap(rhs)</code>.</p>
<p>This overload participates in overload resolution only if <code>lhs.swap(rhs)</code> is valid.</p>
<p>This function is not visible to ordinary <a href="../../language/unqualified_lookup.html" title="cpp/language/unqualified lookup">unqualified</a> or <a href="../../language/qualified_lookup.html" title="cpp/language/qualified lookup">qualified lookup</a>, and can only be found by <a href="../../language/adl.html" title="cpp/language/adl">argument-dependent lookup</a> when <code>std::expected&lt;T, E&gt;</code> is an associated class of the arguments.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> lhs, rhs </td> <td> - </td> <td> <code>expected</code> objects whose states to swap </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>(none)</p>
<h3 id="Exceptions"> Exceptions</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec.html" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code>noexcept(noexcept(lhs.swap(rhs)))</code></span></div>
</div> <h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap.html" title="cpp/utility/expected/swap"> <span class="t-lines"><span>swap</span></span></a></div> </td> <td> exchanges the contents <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/expected/swap2" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/expected/swap2</a>
  </p>
</div>
